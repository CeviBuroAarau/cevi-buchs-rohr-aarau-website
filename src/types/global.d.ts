declare let global: any; // eslint-disable-line @typescript-eslint/no-explicit-any
