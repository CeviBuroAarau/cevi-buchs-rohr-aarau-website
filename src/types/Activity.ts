export interface Activity {
  title: string;
  url: string;
  thumb: string;
}

export interface CockpitActivities {
  entries: ActivityRaw[];
}

export interface ActivityRaw {
  title: string;
  image: {
    path: string;
  };
  thumb: {
    path: string;
  };
}
