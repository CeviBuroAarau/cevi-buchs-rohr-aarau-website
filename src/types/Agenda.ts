export interface Agenda {
  title: string;
  text: string;
  date: Date;
}

export interface CockpitAgenda {
  entries: Agenda[];
}

export interface CockpitScope {
  display: string;
}

export interface CockpitEventInfo {
  scope: CockpitScope;
  text: string;
  date: Date;
}

export interface EventInfo {
  scope: string;
  text: string;
  date: Date;
}

export interface CockpitEventInfos {
  entries: CockpitEventInfo[];
}
