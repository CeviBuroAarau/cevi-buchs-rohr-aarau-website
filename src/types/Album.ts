export interface Album {
  title: string;
  dateString: string;
  date: Date;
  year: number;
  images: AlbumImage[];
  previewImage: AlbumPreviewImage;
}

export interface AlbumImage {
  title: string;
  url: string;
}

export interface AlbumPreviewImage {
  url: string;
}

export interface CockpitAlbum {
  entries: CockpitAlbumEntry[];
}

export interface CockpitAlbumEntry {
  title: string;
  dateString: string;
  date: Date;
  year: number;
  images: CockpitAlbumImage[];
  previewImage: CockpitAlbumPreviewImage;
}

export interface CockpitAlbumImage {
  meta: {
    title: string;
    asset: string;
  };
  path: string;
}

export interface CockpitAlbumPreviewImage {
  path: string;
}
