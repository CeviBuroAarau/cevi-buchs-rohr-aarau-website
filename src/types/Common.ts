export enum FormState {
  NotDisplayed,
  Displayed,
}
