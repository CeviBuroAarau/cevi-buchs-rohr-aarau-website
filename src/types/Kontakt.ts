export interface KontaktFormRequest {
  form: {
    name: string;
    email: string;
    message: string;
  };
}

export interface KontaktFormResponse {
  error: string;
  data: {
    name: string;
    email: string;
    message: string;
  };
}
