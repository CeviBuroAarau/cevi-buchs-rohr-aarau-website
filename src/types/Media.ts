export interface Media {
  type: string;
  date: Date;
  description: string;
  file: string;
}

export interface CockpitMedia {
  entries: Media[];
}
