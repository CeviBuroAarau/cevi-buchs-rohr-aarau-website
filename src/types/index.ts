export * from "./Activity";
export * from "./Agenda";
export * from "./Album";
export * from "./Media";
