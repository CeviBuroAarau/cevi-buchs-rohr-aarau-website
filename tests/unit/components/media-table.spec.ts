import { shallowMount } from "@vue/test-utils";
import MediaTable from "@/components/media-table.vue";
import formatDateFilter from "@/filters/DateFilter.ts";
import { Media } from "@/types/Media";

const media: Media[] = [
  {
    type: " news",
    date: new Date(2021, 0, 1),
    description: "Some text",
    file: "Some file",
  },
];

test("can be instantiated", () => {
  const wrapper = shallowMount(MediaTable, {
    stubs: [],
    propsData: {
      media: media,
    },
  });

  expect(wrapper.isVueInstance).toBeTruthy();
});

test("renders correctly", () => {
  const wrapper = shallowMount(MediaTable, {
    stubs: [],
    propsData: {
      media: media,
    },
    filters: {
      formatDateFilter,
    },
  });

  expect(wrapper.element).toMatchSnapshot();
});
