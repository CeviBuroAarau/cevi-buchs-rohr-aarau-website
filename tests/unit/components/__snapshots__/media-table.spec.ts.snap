// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  aria-describedby="medialist"
  class="table is-bordered is-striped is-narrow is-hoverable is-fullwidth"
>
  <tr>
    <th
      scope="col"
    >
      Datum
    </th>
     
    <th
      scope="col"
    >
      Artikel
    </th>
  </tr>
   
  <tr>
    <td>
      01.01.2021
    </td>
     
    <td>
      <div>
        Some text
      </div>
       
      <br />
       
      <a
        href="Some file"
      >
        Anschauen
      </a>
    </td>
  </tr>
</table>
`;
